hazelcast/src/main/java/com/hazelcast/internal/probing/ProbingCycle.java

```diff
@@ -0,0 +1,222 @@
+     * field or method name or {@link Probe#name()} (if given). The metric type
+     * (long or double) is derived from the field type or method return type. The
+     * set of supported types with a known mapping is fix and implementation
+     * dependent.
+     * 
+     * The method exits with the same {@link Tags} context it is called with.
+     * 
+     * @param instance a obj with fields or methods annotated with {@link Probe}
+     */
+    void probe(Object instance);
+
+    /**
+     * Default method to call {@link #probe(Object)} for an array of instances. This
+     * is mostly useful for types that implement {@link Tagging} and thereby provide
+     * their own context.
+     * 
+     * The method exits with the same {@link Tags} context it is called with.
+     * Instances do not affect each others context.
+     * 
+     * @param instances may be null (no effect)
+     */
+    void probe(Object[] instances);
+
+    /**
+     * Similar to {@link #probe(Object)} just that all names become
+     * {@code <prefix>.<name>} instead of just {@code <name>}.
+     * 
+     * The method exits with the same {@link Tags} context it is called with.
+     * 
+     * @param prefix prefix to use for all names (provided without dot)
+     * @param instance a obj with fields or methods annotated with {@link Probe}
+     */
+    void probe(CharSequence prefix, Object instance);
+
+    /**
+     * Similar to {@link #probe(Object)} just that probed methods are not identified
+     * using annotations but given as a list. All methods use the specified level.
+     * No fields will be probed.
+     * 
+     * This can be used to probe beans that cannot be annotated. To prefix this
+     * {@link Tags#prefix(CharSequence)} can be used. Remember though that the
+     * prefix will remain in the context.
+     * 
+     * @param level the level to use for all probes created
+     * @param instance the obj whose methods to probe.
+     * @param methods the names of the methods to probe. Methods that do not exist
+     *        or have the wrong number of arguments (not none) are ignored.
+     */
+    void probe(ProbeLevel level, Object instance, String... methods);
+
+    void probe(CharSequence name, long value);
+
+    void probe(CharSequence name, double value);
+
+    void probe(CharSequence name, boolean value);
+
+    void probe(ProbeLevel level, CharSequence name, long value);
+
+    void probe(ProbeLevel level, CharSequence name, double value);
+
+    void probe(ProbeLevel level, CharSequence name, boolean value);
+
+    /**
+     * Used to forward metrics received from the client. This is only different from
+     * {@link #probe(CharSequence, long)} except that it will not escape the name
+     * but un-escaping escaped backslashes.
+     * 
+     * @param name a properly escaped name (as written before)
+     * @param value value as written before (double already converted to long)
+     */
+    void probeForwarded(CharSequence name, long value);
```

Comment:
Should this not be a setting in the probe renderer; why should the probe cycle care?

Reply:
It has to do with escaping. names already rendered by the client are already escaped and escaping one more time is wrong. un-escaping names from client again would cause litter. Also the client adds a delimiter what requires extra escaping that delimiter as well what is undone here. So the who thing is just required because we use `String` to transport all metrics data from the client. I did not want to change message protocols but I prepared around it so we can at some point use a better fitting type when sending/receiving client metrics.